presto-hive/src/main/java/com/facebook/presto/hive/StoragePartitionLoader.java

```diff
@@ -300,14 +301,16 @@ public ListenableFuture<?> loadPartition(HivePartitionMetadata partition, HiveSp
         if (shouldUseFileSplitsFromInputFormat(inputFormat, directoryLister)) {
             return handleGetSplitsFromInputFormat(configuration, path, schema, inputFormat, stopped, hiveSplitSource, splitFactory);
         }
+        // S3 Select pushdown works at a granularity finer than individual S3 objects for uncompressed files
+        boolean isSelectSplittable = S3SelectPushdown.isSplittable(s3SelectPushdownEnabled, schema, inputFormat, path);
```

Comment:
It feels like this check actually belongs in `InternalHiveSplitFactory#createInternalHiveSplit`, because the `Path` argument here is actually the table root and not an individual file path. That is to say, that you actually need to check the compression codec on each individual file (since they may not necessarily be the same) and you also can avoid doing (slightly) more expensive checks that aren't necessary like when the file is too small to benefit from splitting anyway. Additionally, the logic there is going to re-check with the `InputFormat#isSplittable` which for `TextInputFormat` is not appropriate for the constraints around S3 select.

Reply:
This was previously addressed.